pkgs/applications/misc/cpp-ethereum/default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, jsoncpp
+, libjson_rpc_cpp
+, curl
+, boost
+, leveldb
+, cryptopp
+, libcpuid
+, opencl-headers
+, ocl-icd
+, miniupnpc
+, libmicrohttpd
+, gmp
+, mesa
+, extraCmakeFlags ? []
+}:
+stdenv.mkDerivation rec {
+  name = "cpp-ethereum-${version}";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ethereum";
+    repo = "cpp-ethereum";
+    rev = "62ab9522e58df9f28d2168ea27999a214b16ea96";
+    sha256 = "1fxgpqhmjhpv0zzs1m3yf9h8mh25dqpa7pmcfy7f9qiqpfdr4zq9";
+  };
+
+  cmakeFlags = with stdenv.lib; concatStringsSep " " ([ "-DCMAKE_BUILD_TYPE=Release" ] ++ extraCmakeFlags);
+
+  configurePhase = ''
+    export BOOST_INCLUDEDIR=${boost.dev}/include
+    export BOOST_LIBRARYDIR=${boost.out}/lib
+
+    mkdir -p Build/Install
+    pushd Build
+
+    cmake .. -DCMAKE_INSTALL_PREFIX=$(pwd)/Install $cmakeFlags
+  '';
+
+  buildPhase = ''
+    make -j $(nproc)
```

Comment:
I'd think this would be unnecessary. In any case, it'd be better to respect `$NIX_BUILD_CORES` instead of calling `nproc` here.